Tests/StoreKitUnitTests/ProductsManagerTests.swift

```diff
@@ -126,3 +128,75 @@ class ProductsManagerTests: StoreKitConfigTestCase {
     }
 
 }
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class ProductsManagerDiagnosticsTrackingTests: ProductsManagerTests {
+
+    private var mockDiagnosticsTracker: MockDiagnosticsTracker!
+
+    private var productsManager: ProductsManager!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        self.mockDiagnosticsTracker = MockDiagnosticsTracker()
+    }
+
+    func testFetchProductsWithIdentifiersSK1TracksCorrectly() throws {
+        let manager = self.createManager(storeKitVersion: .storeKit1,
+                                         diagnosticsTracker: self.mockDiagnosticsTracker)
+
+        let identifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        _ = waitUntilValue(timeout: Self.requestDispatchTimeout) { completed in
+            manager.products(withIdentifiers: Set([identifier]), completion: completed)
+        }
+
+        expect(self.mockDiagnosticsTracker.trackedProductsRequestParams.count) == 1
+        let params = self.mockDiagnosticsTracker.trackedProductsRequestParams.first
+        expect(params?.wasSuccessful) == true
+        expect(params?.storeKitVersion) == .storeKit1
+        expect(params?.errorMessage) == nil
+        expect(params?.errorCode) == nil
+    }
+
+    func testFetchProductsWithIdentifiersSK2TracksCorrectly() throws {
+        let manager = self.createManager(storeKitVersion: .storeKit2,
+                                         diagnosticsTracker: self.mockDiagnosticsTracker)
+
+        let identifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        _ = waitUntilValue(timeout: Self.requestDispatchTimeout) { completed in
+            manager.products(withIdentifiers: Set([identifier]), completion: completed)
+        }
+
+        expect(self.mockDiagnosticsTracker.trackedProductsRequestParams.count) == 1
+        let params = self.mockDiagnosticsTracker.trackedProductsRequestParams.first
+        expect(params?.wasSuccessful) == true
+        expect(params?.storeKitVersion) == .storeKit2
+        expect(params?.errorMessage) == nil
+        expect(params?.errorCode) == nil
+    }
+
+    #if swift(>=5.9)
+    @available(iOS 17.0, tvOS 17.0, macOS 14.0, watchOS 10.0, *)
+    func testFetchProductsWithIdentifiersSK2ErrorTracksCorrectly() async throws {
```

Comment:
I only added an error test case for SK2 since `setSimulatedError` didn't seem to work well for SK1 (the test got stuck). Do lmk if I missed something!